clients/pkg/promtail/targets/file/tailer.go

```diff
@@ -149,15 +165,26 @@ func (t *tailer) readLines() {
 			continue
 		}
 
+		var text string
+		if t.decoder != nil {
+			var err error
+			text, err = t.convertToUTF8(line.Text)
+			if err != nil {
+				level.Error(t.logger).Log("msg", "failed to convert encoding", "error", err)
+				t.metrics.encodingFailures.WithLabelValues(t.path).Inc()
```

Comment:
I'm wondering how we should handle this error case.

Thoughts: 
* we could write the error into the log line and send that to Loki?
* we could send the log line as received on to Loki even though it didn't format?

I also wonder if we should log at Error level or Debug level, if this happens a lot it in a way someone can't control but they still want all the lines they can get then this could be very noisy, on the other hand if we only increment the metric people may not notice that and not realize they are missing logs.

Reply:
> we could write the error into the log line and send that to Loki?
I like that tbh. I'll modify this to send it.

> we could send the log line as received on to Loki even though it didn't format?
Loki won't be able to parse it so it won't be that useful. I feel like the best experience for users regarding encoding is to be really verbose, so that they can easily know that something isn't properly configured encoding-wise, but once it isn't verbose anymore they can be sure things are working as expected. Another thing is that this encoding won't do anything by default, so we're only bringing noise to future users that are willing to use this.

> I also wonder if we should log at Error level or Debug level, if this happens a lot it in a way someone can't control but they still want all the lines they can get then this could be very noisy, on the other hand if we only increment the metric people may not notice that and not realize they are missing logs.
Yep it has this trade-off. Personally, I prefer it with error level because this looks important enough to be listed as an error but it is debatable.

Nonetheless, I feel like we should first focus on users that:
- Have their data very consistent towards one encoding format
- Really require us to support encoding
This way, edge cases like users having a mix of log lines encoding etc can be addressed in the future whenever we find there's enough demand for it.